adal/src/main/java/com/microsoft/aad/adal/AcquireTokenSilentHandler.java

```diff
@@ -128,7 +129,32 @@ AuthenticationResult acquireTokenWithRefreshToken(final String refreshToken)
                 Logger.i(TAG, "Refresh token is not returned or empty", "");
                 result.setRefreshToken(refreshToken);
             }
-        } catch (final IOException | AuthenticationException exc) {
+        } catch (final AuthenticationException exc) {
+            //if get 503,504,500 && outageMode is on, return the stale token
+            if (exc.getCode().equals(ADALError.SERVER_NOT_RESPONDING) && mOutageModeIsOn) {
+                final TokenCacheItem accessTokenItem = mTokenCacheAccessor.getRegularRefreshTokenCacheItem(mAuthRequest.getResource(),
```

Comment:
can you please move the code in this catch block to another function and call it?
I want to see something like
if (shouldTryExtendedLifetime(exc.getCode())) {
    return createExtendedLifeTimeResult()
}

Line 143-151 can remain unchanged

---
In reply to: [70705480](https://github.com/AzureAD/azure-activedirectory-library-for-android/pull/676#discussion_r70705480) [](ancestors = 70705480)